rust/src/jsonbuilder.rs

```diff
@@ -502,6 +502,20 @@ impl JsonBuilder {
         Ok(self)
     }
 
+    /// Set a key and a string value on an object, with a limited size
+    pub fn set_string_limited(
+        &mut self, key: &str, val: &str, limit: usize,
+    ) -> Result<&mut Self, JsonError> {
+        if val.len() > limit {
+            let valtrunc = &val[..limit];
```

Comment:
Is it not a problem with UTF-8 or something ?
Like you panic if you split in a middle of a UTF char...

Reply:
From https://doc.rust-lang.org/std/string/struct.String.html

> This is also called “string slicing”. Note this will panic if the byte indices provided are not character boundaries

Reply:
Ah, TIL. Totally, yes. I assumed that Rust would slice a string by runes, not bytes. Will adapt that.